core/kernel/dt.c

```diff
@@ -472,6 +496,203 @@ void *get_secure_dt(void)
 	return fdt;
 }
 
+#ifdef CFG_DT_CACHED_NODE_INFO
+/* Reference to the FDT for which node information are cached */
+static const void *cached_node_info_fdt;
+
+/*
+ * struct parent_node_cache - Cache of FDT parent node references
+ *
+ * @node_offset: Offset of a node in @cached_node_info_fdt
+ * @parent_offset: Offset of @node_offset parent node or -1 if root node
+ * @address_cells: Parent node #address-cells property value or 0
+ * @size_cells: Parent node #size-cells property value or 0
+ */
+struct parent_node_cache {
```

Comment:
I find the `parent` part in the name confusing since the struct describes a node. How about `struct cached_node`?

Reply:
The struct intended to relate to parent node information of a node, hence this name.
I'll fix that to use a common struct for both parent node info and node phandle info. I'll use `cached_node`.